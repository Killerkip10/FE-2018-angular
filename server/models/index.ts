export {User} from './user';
export {Login} from './login';
export {ShortUser} from './short-user';
