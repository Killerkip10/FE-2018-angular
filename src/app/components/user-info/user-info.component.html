<mat-card>
  <mat-card-title>{{userInfo?.name}}</mat-card-title>
  <mat-card-content>
    <p>{{userInfo?.login}}</p>
    <p>{{userInfo?.information}}</p>
    <p>{{userInfo?.age}}</p>
    <p>{{userInfo?.dateOfBirth}}</p>
    <p>{{userInfo?.dateOfFirstLogin}}</p>
    <p>{{userInfo?.dateOfNextNot}}</p>
  </mat-card-content>
</mat-card>
