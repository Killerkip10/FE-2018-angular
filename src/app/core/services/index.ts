export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
