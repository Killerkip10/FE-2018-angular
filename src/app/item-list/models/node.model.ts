export interface Node {
  value: string;
  open: boolean;
  children: Node[];
}
