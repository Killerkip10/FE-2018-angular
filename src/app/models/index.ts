export * from '../../../server/models/index';
