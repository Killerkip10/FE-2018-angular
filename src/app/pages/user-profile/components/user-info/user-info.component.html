<mat-card>
  <mat-toolbar>
    <span>{{userInfo?.name}}</span>
    <span>{{userInfo?.login}}</span>
  </mat-toolbar>

  <mat-card-content>
    <mat-card-subtitle>{{userInfo?.information}}</mat-card-subtitle>
    <p>{{userInfo?.age}}</p>
    <p>{{userInfo?.dateOfBirth}}</p>
    <p>{{userInfo?.dateOfFirstLogin}}</p>
    <p>{{userInfo?.dateOfNextNot}}</p>
  </mat-card-content>
</mat-card>
