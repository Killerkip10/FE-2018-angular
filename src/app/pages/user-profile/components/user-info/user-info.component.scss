mat-card{
  padding: 0;
  mat-toolbar{
    display: flex;
    justify-content: space-between;
    >:last-child{
      color: rgba(0,0,0,.54);
    }
  }
  mat-card-content {
    padding: 16px;
  }
}
