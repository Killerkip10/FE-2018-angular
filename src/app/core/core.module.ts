import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
