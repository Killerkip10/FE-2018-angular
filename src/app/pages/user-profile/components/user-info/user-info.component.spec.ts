import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';

import {UserInfoComponent} from './user-info.component';
import {User} from '../../../../models';

describe('UserInfoComponent', () => {
  let component: UserInfoComponent;
  let fixture: ComponentFixture<UserInfoComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        UserInfoComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserInfoComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.userInfo = {} as User;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.userInfo = {
      name: 'Vlad',
      login: '1234',
      information: 'i like angular',
      age: '24',
      dateOfNextNot: '2018-11-10T18:52:23.023Z',
      dateOfFirstLogin: '2018-11-10T18:52:23.023Z',
      dateOfBirth: '2018-11-10T18:52:23.023Z'
    } as User;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
