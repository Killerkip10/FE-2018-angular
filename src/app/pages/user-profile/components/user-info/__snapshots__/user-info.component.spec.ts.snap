// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoComponent snapshot 1`] = `
<app-user-info>
  <mat-card>
    <mat-toolbar>
      <span>
        
      </span>
      <span>
        
      </span>
    </mat-toolbar>
    <mat-card-content>
      <mat-card-subtitle>
        
      </mat-card-subtitle>
      <p>
        
      </p>
      <p>
        
      </p>
      <p>
        
      </p>
      <p>
        
      </p>
    </mat-card-content>
  </mat-card>
</app-user-info>
`;

exports[`UserInfoComponent snapshot 2`] = `
<app-user-info
  userInfo={[Function Object]}
>
  <mat-card>
    <mat-toolbar>
      <span>
        
      </span>
      <span>
        
      </span>
    </mat-toolbar>
    <mat-card-content>
      <mat-card-subtitle>
        
      </mat-card-subtitle>
      <p>
        
      </p>
      <p>
        
      </p>
      <p>
        
      </p>
      <p>
        
      </p>
    </mat-card-content>
  </mat-card>
</app-user-info>
`;

exports[`UserInfoComponent snapshot 3`] = `
<app-user-info
  userInfo={[Function Object]}
>
  <mat-card>
    <mat-toolbar>
      <span>
        Vlad
      </span>
      <span>
        1234
      </span>
    </mat-toolbar>
    <mat-card-content>
      <mat-card-subtitle>
        i like angular
      </mat-card-subtitle>
      <p>
        24
      </p>
      <p>
        2018-11-10T18:52:23.023Z
      </p>
      <p>
        2018-11-10T18:52:23.023Z
      </p>
      <p>
        2018-11-10T18:52:23.023Z
      </p>
    </mat-card-content>
  </mat-card>
</app-user-info>
`;
